pkg/arvo/gen/hood/merge/help.txt

```diff
@@ -0,0 +1,110 @@
+Usage:
+
+    |merge %destination-desk ~source-ship %source-desk
+    |merge %destination-desk ~source-ship %source-desk, =gem %strategy
+    |merge %destination-desk ~source-ship %source-desk, =cas ud+5
+
+We support various merge strategies.  A "commit" is a snapshot of
+the files with a list of parents plus a date.   Most commits have
+one parent; a "merge" commit is a commit with two parents.  The
+%home desk starts with an initial commit with no parents; commits
+with several parents ("octopus merges") are possible but we don't
+generate them right now.
+                                                                      
+Unless otherwise specified, all of the following create a new commit
+with the source and destination commits as parents.
+                                                                      
+Several strategies need a "merge-base".  They find it by identifying
+the most recent common ancestor of the two desks.  If none, fail
+with %merge-no-merge-base; if there are two or more, pick one.
+                                                                      
+%init: the only way to create a desk.  Not a true merge, since it
+simply assigns the source commit to the destination.
+                                                                      
+%fine: if source or destination are in the ancestry of each other,
+use the newer one; else abort.  If the destination is ahead of the
+source, succeed but do nothing.  If the source is ahead of the
+destination, assign the next revision number to the source commit.
+Some call this "fast-forward".
+                                                                      
+%meet: combine changes, failing if both sides changed the same file.
+Specifically, take diff(merge-base,source) and
+diff(merge-base,destination) and combine them as long as those diffs
+touch different files.
+                                                                      
+%mate: combine changes, failing if both sides changed the same part
+of a file.  Identical to %meet, except that some marks, like %hoon,
+allow intelligent merge of changes to different parts of a file.
+                                                                      
+%meld: combine changes; if both sides changed the same part of a
```

Comment:
Does this mean %meld doesn't apply either change, and just goes with the old version?

Reply:
That's correct.  This makes it convenient to construct a 3-way diff later on, but it's somewhat non-intuitive

Reply:
Ah, I see, this is used for annotating conflicts on top of it.  Makes sense.